src/test/java/com/cloudbees/jenkins/plugins/bitbucket/UriResolverTest.java

```diff
@@ -51,23 +53,26 @@ public void httpUriResolver() throws Exception {
                 "user1",
                 "repo1"
         ));
-        api = new BitbucketServerAPIClient("http://localhost:1234", "test", null, (BitbucketAuthenticator) null, false);
+        api = new BitbucketServerAPIClient("http://localhost:1234", "test", null, null, false,
+                BitbucketServerWebhookImplementation.PLUGIN);
```

Comment:
Removing the `BitbucketServerWebhookImplementation.PLUGIN` from unit tests definitely creates the null pointer even with a default set 😅

Reply:
This definitely produces another NPE related to the test setup, since there's no endpoint configuration at all available:
```
java.lang.NullPointerException
	at com.cloudbees.jenkins.plugins.bitbucket.endpoints.BitbucketServerEndpoint.findWebhookImplementation(BitbucketServerEndpoint.java:106)
	at com.cloudbees.jenkins.plugins.bitbucket.server.client.BitbucketServerAPIClient.<init>(BitbucketServerAPIClient.java:199)
	at com.cloudbees.jenkins.plugins.bitbucket.UriResolverTest.httpUriResolver(UriResolverTest.java:56)
```